erizo/src/erizo/bandwidth/ConnectionQualityCheck.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * ConnectionQualityCheck.cpp
+ */
+#include "bandwidth/ConnectionQualityCheck.h"
+
+#include <algorithm>
+
+#include "MediaStream.h"
+#include "rtp/RtpUtils.h"
+
+namespace erizo {
+
+DEFINE_LOGGER(ConnectionQualityCheck, "bandwidth.ConnectionQualityCheck");
+
+constexpr uint8_t ConnectionQualityCheck::kHighAudioFractionLostThreshold;
+constexpr uint8_t ConnectionQualityCheck::kLowAudioFractionLostThreshold;
+
+ConnectionQualityCheck::ConnectionQualityCheck()
+    : quality_level_{ConnectionQualityLevel::GOOD}, buffer_{1} {
+}
+
+void ConnectionQualityCheck::onFeedback(std::shared_ptr<DataPacket> packet,
+    const std::vector<std::shared_ptr<MediaStream>> &streams) {
+  if (streams.size() != buffer_.capacity()) {
+    buffer_.set_capacity(streams.size());
+  }
+  int reports_count = 0;
+  RtpUtils::forEachRtcpBlock(packet, [streams, this, &reports_count](RtcpHeader *chead) {
+    reports_count++;
+    uint32_t ssrc = chead->isFeedback() ? chead->getSourceSSRC() : chead->getSSRC();
+    bool is_rr = chead->isReceiverReport();
+    if (is_rr) {
+      ELOG_WARN("  RR Block_COUNT %d", chead->getBlockCount());
+    } else {
+      ELOG_WARN("  %d", chead->getPacketType());
+    }
+    uint8_t fraction_lost = is_rr ? chead->getFractionLost() : 0;
+    std::for_each(streams.begin(), streams.end(),
+        [ssrc, fraction_lost, is_rr, this] (const std::shared_ptr<MediaStream> &media_stream) {
+      bool is_audio = media_stream->isAudioSourceSSRC(ssrc) || media_stream->isAudioSinkSSRC(ssrc);
```

Comment:
While I think using audio loss is a good strategy, this would invalidate this mechanism in all streams with muted audio.
Consider evaluating fraction_loss also for video ssrcs. A possible compromise would be to assign a higher weight to audio loss when computing the mean.